sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -900,6 +902,18 @@ class Analyzer(
             }
           case _ => u
         }
+
+      case u @ UnresolvedTableOrView(identifier) =>
+        expandRelationName(identifier) match {
+          case SessionCatalogAndIdentifier(catalog, ident) =>
+            CatalogV2Util.loadTable(catalog, ident) match {
+              // We don't support view in v2 catalog yet. Here we treat v1 view as table and use
+              // `ResolvedTable` to carry it.
+              case Some(table) => ResolvedTable(catalog.asTableCatalog, ident, table)
```

Comment:
Do you mean to add a case?

```
CatalogV2Util.loadTable(catalog, ident) match {
  case Some(v1Table: V1Table) if v1Table.v1Table.tableType == CatalogTableType.VIEW =>
  ...
}
```

Otherwise I'm not sure why `case Some(table) =>` is for v1 view.

Reply:
good point. Adding a case is clearer.